pkg/util/glog/glog.go

```diff
@@ -1,101 +1,224 @@
 }
 
-func (glogger) V(level int) Logger {
-	return gverbose{glog.V(glog.Level(level))}
+func (glogger) V(level int32) VerboseLogger {
+	return gverbose{log.V(log.Level(level))}
 }
 
 func (glogger) Infof(format string, args ...interface{}) {
-	glog.Infof(format, args...)
+	log.Infof(format, args...)
 }
 
-// gverbose handles glog.V(x) calls
-type gverbose struct {
-	glog.Verbose
+func (glogger) Info(args ...interface{}) {
+	log.Info(args...)
 }
 
-func (gverbose) Is(level int) bool {
-	return bool(glog.V(glog.Level(level)))
+func (glogger) Warningf(format string, args ...interface{}) {
+	log.Warningf(format, args...)
 }
 
-func (gverbose) V(level int) Logger {
-	if glog.V(glog.Level(level)) {
-		return Log
-	}
-	return None
+func (glogger) Warning(args ...interface{}) {
+	log.Warning(args...)
+}
+
+func (glogger) Errorf(format string, args ...interface{}) {
+	log.Errorf(format, args...)
+}
+
+func (glogger) Error(args ...interface{}) {
+	log.Error(args...)
+}
+
+func (glogger) Fatalf(format string, args ...interface{}) {
+	log.Fatalf(format, args...)
+}
+
+func (glogger) Fatal(args ...interface{}) {
+	log.Fatal(args...)
+}
+
+// gverbose handles glog.V(x) calls
+type gverbose struct {
+	log.Verbose
 }
 
 func (g gverbose) Infof(format string, args ...interface{}) {
 	g.Verbose.Infof(format, args...)
 }
 
-// file logs the provided messages at level or below to the writer, or delegates
+func (g gverbose) Info(args ...interface{}) {
+	g.Verbose.Info(args...)
+}
+
+// FileLogger logs the provided messages at level or below to the writer, or delegates
 // to glog.
-type file struct {
-	w     io.Writer
-	level int
+type FileLogger struct {
+	w     *bufio.Writer
+	level int32
 }
 
-func (f file) Is(level int) bool {
-	return level <= f.level
+// Is returns whether the current logging level is greater than or equal to the parameter
+func (f *FileLogger) Is(level int32) bool {
+	return level <= atomic.LoadInt32(&f.level)
 }
 
-func (f file) V(level int) Logger {
+// V will returns a logger which will discard output if the specified level is greater than the current logging level
+func (f *FileLogger) V(level int32) VerboseLogger {
 	// only log things that glog allows
-	if !glog.V(glog.Level(level)) {
+	if !log.V(log.Level(level)) {
 		return None
 	}
+
 	// send anything above our level to glog
-	if level > f.level {
+	if level > atomic.LoadInt32(&f.level) {
 		return Log
 	}
 	return f
 }
 
-func (f file) Infof(format string, args ...interface{}) {
-	fmt.Fprintf(f.w, format, args...)
-	if !strings.HasSuffix(format, "\n") {
-		fmt.Fprintln(f.w)
+func (f *FileLogger) writeln(line string) {
+	defer f.w.Flush()
+	if atomic.LoadInt32(&f.level) >= 100 {
+		pc := make([]uintptr, 10)
+		runtime.Callers(4, pc)
+		function := runtime.FuncForPC(pc[0])
+		file, line := function.FileLine(pc[0])
+		s := strings.Split(file, "/") // remove absolute path from file
+		file = s[len(s)-1]
+		f.w.WriteString(fmt.Sprintf("[%s:%d] ", file, line-1))
```

Comment:
I left it in pending the answer to the glog usage question. If we want *real* filenames and line numbers for verbose logging statements, we can't defer to golog since it will only report pkg/util/glog/glog.go as the source of logging statements.